src/clone.c

```diff
@@ -53,6 +56,76 @@
 # define ARG_CTID	4
 #endif
 
+static bool show_namespace = false;
+extern void namespace_auxstr_init(void)
+{
+	show_namespace = true;
+}
+
+/* derrived from print_dirfd */
+static const char *
+read_namespace_id(int pid, const char *const ns_type)
+{
+	static const char ns_path[] = "/proc/%d/ns/%s";
+	/* "cgroup" is the longest name in the dentries in /proc/$pid/ns. */
+	char linkpath[sizeof(ns_path) + sizeof(int) * 3 + sizeof("cgroup")];
+	xsprintf(linkpath, ns_path, pid, ns_type);
+
+	static char buf[PATH_MAX + 1];
+	ssize_t n = readlink(linkpath, buf, sizeof(buf));
+	if ((size_t) n >= sizeof(buf))
+		return NULL;
+	buf[n] = '\0';
+	return buf;
+}
+
+static const char *
+get_namespace_auxstr(int pid, uint64_t flags)
+{
+	static char str[sizeof("cgroup:[4026531835], "
+			          "ipc:[4026531839], "
+			          "mnt:[4026531841], "
+			          "net:[4026531840], "
+			          "pid:[4026531836], "
+			         "time:[4026531834], "
+			         "user:[4026531837], "
+			          "uts:[4026531838]" )
+			/* for the last null char */
+			+ 1];
+	char *p = str;
+
+#define APPEND(NS_TYPE) do {				\
+		const char *ns_name = read_namespace_id(pid, NS_TYPE); \
+		if (ns_name == NULL)				       \
+			break;				\
+		if (str != p)				\
+			p = xappendstr(str, p, ", ");	\
+		p = xappendstr(str, p, "%s", ns_name);	\
+	} while (0)
+
+	*p = '\0';
+	if (flags & CLONE_NEWCGROUP)
+		APPEND("cgroup");
```

Comment:
Note that clone3's CLONE_INTO_CGROUP is also "cgroup".

Reply:
fixed locally.